e2e-test/reconcile_namespace_test.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+set -eo pipefail
+
+test_dir=e2e-test
+
+source ${test_dir}/functions.sh
+
+e2e-test-setup
+
+WATCH_NAMESPACE=espejo start-espejo
+
+sleep 5s
+echo "- Creating sync config in watched namespace"
+kubectl -n espejo apply -f ${test_dir}/syncconfig.yaml
+sleep 5s
+
+echo "- Verifying that there is an object in another namespace"
+kubectl get cm -n e2e-test | grep "espejo-e2e-test-data"
```

Comment:
Instead of waiting a fixed amount of time with `sleep` you could use [kubectl wait](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#wait) with a timeout. I think this line would look something like this
```suggestion
kubectl -n e2e-test wait --for condition=available --timeout=15s configmap/espejo-e2e-test-data
```

(and the `sleep 5` can be removed)

Reply:
Even with a timeout of 15s, if there's no config map, it immediately returns:
```
$ kubectl -n e2e-test wait --for condition=available --timeout=15s configmap/espejo-e2e-test-data
Error from server (NotFound): configmaps "espejo-e2e-test-data" not found
```

Reply:
I don't think you can use `wait` on ConfigMaps, because ConfigMaps don't implement a `condition` field in their status  (they don't implement *any* `.status` for that matter) like pods do for example